### DISCUSSION TITLE
Canvas2D multiple readback operations getImageData willReadFrequently

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#7623] [Q&A]
I started noticing with a recent version of Chrome (v108) the below warning within the console...

`Canvas2D: Multiple readback operations using getImageData are faster with willReadFrequently attribute set to true.`

The messages are coming from deck (v8.8.23) or deck dependencies...
- text-layer/font-atlas-manager
- mapbox/tiny-sdf

Curious if this has been reported previously? Seems like it may be impactful to performance and has been reported to other projects (https://github.com/openlayers/openlayers/issues/14221).

@Pessimistress @ibgreen

**Comment by ibgreen** [comment]
Without looking at the code, I would expect that adding that flag to any `canvas.getContext()` call in the font-atlas-manager should be a quick one-line PR. I am not sure it will have a noticeable impact as reading images from the canvas is more of an initialization step, but it would be nice to get rid of the warning and it shouldn't hurt (unless there are some complications, like other browsers throwing errors if that option is provided etc).

The tiny-sdf issue of should probably be discussed in the mapbox-gl-js repo.

**Comment by jfuehner** [comment]
Sounds like mapbox-gl/tiny-sdf has implemented a fix for this issue https://github.com/mapbox/tiny-sdf/issues/51

**Comment by ibgreen** [comment]
Care to make a similar PR for deck? :)
